frontend/app/services/SubscriptionService.scala

```diff
@@ -140,13 +140,22 @@ class SubscriptionService(val zuoraSoapService: ZuoraSoapService,
   val membershipProductType = "Membership"
   val productRatePlanChargeModel = "FlatFee"
 
+  val productRatePlanTiers = List(FriendTierPlan, StaffPlan,
+    PaidTierPlan(Supporter, true), PaidTierPlan(Supporter, false),
+    PaidTierPlan(Partner, true), PaidTierPlan(Partner, false),
+    PaidTierPlan(Patron, true), PaidTierPlan(Patron, false))
+
   def membershipProducts = zuoraRestService.productCatalogSupplier.get().map(_.productsOfType(membershipProductType))
 
-  def tierRatePlanId(productRatePlan: ProductRatePlan): Future[String] =
-    membershipProducts.map(extractRatePlanIdFromCatalog(productRatePlan))
+  val productRatePlanIdSupplier = new FutureSupplier[Map[ProductRatePlan, String]](
+    for {
+      membershipProductCatalog <- membershipProducts
+    } yield (for {
+      productPlan <- productRatePlanTiers
```

Comment:
I think using the `for` comprehension here makes code less readable and more verbose than using `map`